config.js

```diff
@@ -105,51 +103,61 @@ function exportInitialConfigUnderscored(rankedUnrankedUnderscored) {
     exports[`allow_all_timecontrols${rankedUnrankedUnderscored}`] = false;
 }
 
+function exportInitialConfig() {
+    exportInitialConfigUnderscored("");
+    exportInitialConfigUnderscored("_ranked");
+    exportInitialConfigUnderscored("_unranked");
+
+    exports.start_date = new Date();
+}
+
 function testRankedUnrankedOptions(rankedUnrankedOptions, argv) {
     for (const option of rankedUnrankedOptions) {
         const [general, ranked, unranked] = getArgNamesGRU(option.name);
         
         // check undefined specifically to handle valid values such as 0 or null which are tested false
         if (argv[general] !== undefined) {
             if (argv[ranked] !== undefined) {
-                throw `Cannot use --${general} and --${ranked} at the same time.\nFor ranked games, use either --${general} or --${ranked}, or do not use anything to switch to default settings for this option.`;
+                throw new Error(`Cannot use --${general} and --${ranked} at the same time.\nFor ranked games, use either --${general} or --${ranked}, or do not use anything to switch to default settings for this option.`);
```

Comment:
If we're going to throw error objects, it might be better to extewnd the error object, and later when we catche it, use `instanceof` to check if it's our own custom error.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error#Custom_Error_Types

Reply:
oh very interesting, it is where i currently stopped at the js lesson
i'll try it then 💯